prov/tcp/src/tcpx_ep.c

```diff
@@ -405,6 +405,79 @@ static int tcpx_ep_shutdown(struct fid_ep *ep, uint64_t flags)
 	return FI_SUCCESS;
 }
 
+static int tcpx_pep_sock_create(struct tcpx_pep *pep)
+{
+	struct addrinfo hints, *result, *iter;
+	char sa_ip[INET6_ADDRSTRLEN] = {0};
+	char sa_port[NI_MAXSERV] = {0};
+	int ret;
+
+	memset(&hints, 0, sizeof(hints));
+	hints.ai_family = AF_INET;
+	hints.ai_socktype = SOCK_STREAM;
+	hints.ai_flags = AI_PASSIVE;
+
+	switch (pep->info.addr_format) {
+	case FI_SOCKADDR:
+	case FI_SOCKADDR_IN:
+	case FI_SOCKADDR_IN6:
+		ret = getnameinfo(pep->info.src_addr,
+				  (socklen_t) pep->info.src_addrlen,
+				  sa_ip, INET_ADDRSTRLEN, sa_port,
+				  NI_MAXSERV, 0);
+		if (ret) {
+			FI_WARN(&tcpx_prov, FI_LOG_EP_CTRL,
+				"getnameinfo failed\n");
+			return ret;
+		}
+		break;
+	default:
+		FI_WARN(&tcpx_prov, FI_LOG_EP_CTRL,
+			"invalid source address format\n");
+		return -FI_EINVAL;
+	}
+
+	ret = getaddrinfo(sa_ip, sa_port, &hints, &result);
+	if (ret) {
+		FI_WARN(&tcpx_prov, FI_LOG_EP_CTRL,"getaddrinfo failed");
+		return -FI_EINVAL;
+	}
```

Comment:
The pep->info.src_addr is a sockaddr.  You should be able to just allocate the socket and bind to the address. There shouldn't be a need to call getaddrinfo.

Reply:
I' should have found this myself. Thanks